docs/pages/access-controls/access-graph/self-hosted-helm.mdx

```diff
@@ -0,0 +1,220 @@
+
+```yaml
+postgres:
+  secretName: "teleport-access-graph-postgres"
+
+tls:
+  # You may need to change this value if using cert-manager or other means
+  # to provision TLS certificates for your Kubernetes workloads.
+  existingSecretName: "teleport-access-graph-tls"
+
+clusterHostCAs:
+  # Replace this with the certificate retrieved in Step 1.
+  # `clusterHostCAs` is an array of multiline YAML strings, `- |` must not be removed, and the indentation must be kept.
+  - |
+    -----BEGIN CERTIFICATE-----
+    MIIDqjCCApKgAwIBAgIQMIK8/WiQ/rUOrjlmB0IHVTANBgkqhkiG9w0BAQsFADBv
+    <...>
+    -----END CERTIFICATE-----
+```
+
+Finally, deploy the TAG service using Helm:
+
+```code
+$ helm install -n teleport-access-graph -f tag-values.yaml teleport-access-graph teleport/teleport-access-graph
+$ kubectl -n teleport-access-graph rollout status deployment/teleport-access-graph # Wait for the deployment to succeed
+```
+
+## Step 3/4. Update the Teleport Auth Service configuration
+
+Unless you are using a public Certificate Authority to issue the server certificate for TAG,
+the Teleport Auth Service will need a copy of the certificate of the CA that issued the server certificate.
+
+Create a ConfigMap containing the CA certificate as follows:
+
+```code
+$ kubectl -n teleport create configmap teleport-access-graph-ca \
+    --from-file=ca.pem=./certs/ca.crt
+```
+
+Then, update the values for the deployment of [teleport-cluster](../reference/helm-reference/teleport-cluster/) Helm chart as follows:
+
+```yaml
+auth:
+  teleportConfig:
+    # <...>
+
+    # Add a section for configuring the Teleport Access Graph connection.
+    access_graph:
+      enabled: true
+      endpoint: teleport-access-graph.teleport-access-graph.svc.cluster.local:443
+      ca: /var/run/access-graph/ca.pem
+
+# Provide the TAG CA to the Teleport Auth Service as a volume.
+extraVolumes:
+  - name: tag-ca
+    configMap:
+      name: teleport-access-graph-ca
+
+extraVolumeMounts:
+  - name: tag-ca
+    mountPath: /var/run/access-graph
+```
+
+If not using the `teleport-cluster` Helm chart, you will need to do the equivalent changes to your Teleport cluster deployment:
+* Add the `access_graph:` section at the top-level of the YAML config file for the Teleport Auth Service.
+* Mount the created ConfigMap as a volume so that the Auth Service can read the CA certificate.
+
+Finally, redeploy the Helm chart (assuming the values are stored in `values-teleport.yaml`).
+Once the Auth Service changes succeed, restart the Proxy Service.
+
+```code
+$ helm upgrade -n teleport -f values-teleport.yaml --version <version> teleport teleport/teleport-cluster 
+$ kubectl -n teleport rollout status deployment/teleport-auth   # Wait for the deployment to succeed
+$ kubectl -n teleport rollout restart deployment/teleport-proxy 
+$ kubectl -n teleport rollout status deployment/teleport-proxy  # Wait for the deployment to succeed
```

Comment:
People will copy-paste this without thinking. This includes a hardcoded namespace (which is different from the default one we use in the docs) and a hardcoded release name.

I would remove this code entirely.

Also I'm not sure why you are rolling out auth and proxy deployments. The chart should do it for you on configuration change as the deployment pod spec changes (it contains a config hash). Is there a hidden race condition somewhere requiring to rollout the proxies a second time?

Reply:
> This includes a hardcoded namespace (which is different from the default one we use in the docs) and a hardcoded release name.

Interestingly, the [main guide](https://goteleport.com/docs/deploy-a-cluster/helm-deployments/kubernetes-cluster/) uses the `teleport-cluster` namespace, while the [custom config](https://goteleport.com/docs/deploy-a-cluster/helm-deployments/custom/#step-23-setting-up-a-teleport-cluster-with-helm-using-a-custom-config) guide uses `teleport`).

> Also I'm not sure why you are rolling out auth and proxy deployments.
The auth deployment we are not re-rolling, just waiting for the upgrade to complete (which *is* happening due to the config change).

The proxy deployment is due to the fact that proxies only query the TAG feature status (enabled/disabled) from the Auth once, and then caches it indefinitely. The proxy config does not change, but we sadly still need to re-roll it to pick up the change done in Auth 😕 So we still need this step for the user to fully set up and be able to use TAG.

Would it be better if I replaced the hardcoded namespace etc. with `<Var />`-s that people would need to fill in?

Reply:
> Interestingly, the [main guide](https://goteleport.com/docs/deploy-a-cluster/helm-deployments/kubernetes-cluster/) uses the teleport-cluster namespace, while the [custom config](https://goteleport.com/docs/deploy-a-cluster/helm-deployments/custom/#step-23-setting-up-a-teleport-cluster-with-helm-using-a-custom-config) guide uses teleport).

Yeah, the custom config is a portage of the legacy guides and uses `teleport` like them. I think I should delete it now that you can do everything with the `teleportConfig` ovveride in the `standalone` mode. Users don't need `scratch`.

> The proxy config does not change, but we sadly still need to re-roll it to pick up the change done in Auth 😕

Ugh, even doing a proxy config change might not 100% reliably enable TAG as the proxy might connect to an old auth. We only protect against this in version upgrade scenarios.

> Would it be better if I replaced the hardcoded namespace etc. with <Var />-s that people would need to fill in?

I suppose that's the best solution, given that we have to restart proxies.